### DISCUSSION TITLE
PDF Files are blank

### DISCUSSION THREAD

**Discussion by blainarm387** [dompdf/dompdf#3104] [Q&A]
I'm having trouble with version 2.0.1. I've tried running the example shown on the page (OK, slightly modified) as follows:

```
<?php

use Dompdf\Dompdf;

require __DIR__ . '/../vendor/autoload.php';

$dompdf = new Dompdf();
$dompdf->loadHtml('<p>Yo!</p>');
$dompdf->setPaper('A4');
$dompdf->stream();
```

The file gets created and sent to the browser, but it's blank. I've tried it on php versions 8.1 and 7.4, both on Alpine inside a Docker container and a WAMP server. All installations contain the required extensions (ext-dom, ext-mbstring) as well as the GD extension. Is there something I'm missing here?

**Comment by bsweeney** [comment]
Almost there, you have to call `render` before you stream.

```
<?php

use Dompdf\Dompdf;

require __DIR__ . '/../vendor/autoload.php';

$dompdf = new Dompdf();
$dompdf->loadHtml('<p>Yo!</p>');
$dompdf->setPaper('A4');
$dompdf->render();
$dompdf->stream();
```

**Reply by blainarm387** [reply]
Such a dumb mistake on my part. Thanks for clearing that up!

For some reason, I thought the render method was for outputting the PDF directly to the browser as opposed to streaming it for download.

**Reply by bsweeney** [reply]
It happens. There has been a suggestion to make calling render optional (i.e., implied if the PDF is output) which I think we probably will do at some point.
